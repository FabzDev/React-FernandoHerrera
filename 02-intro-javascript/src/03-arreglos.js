

const arreglo = new Array( 5 );
arreglo.push(1);
arreglo.push(2);

const arreglo2 = [];
arreglo2.push(1);
arreglo2.push(2);
arreglo2.push(3);
arreglo2.push(4);


const arreglo3 = new Array();
arreglo3.push(1);
arreglo3.push(2);

const arregloTask = [ ...arreglo2, 5 ]

const mapArreglo2Mult2 = arreglo2.map((n) => n*2)

console.log(arreglo);
console.log(arreglo2);
console.log(arreglo3);
console.log(arregloTask);
console.log(mapArreglo2Mult2);























