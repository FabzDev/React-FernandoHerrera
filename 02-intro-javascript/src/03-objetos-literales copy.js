

const persona = {
    nombre: "Eren",
    apellido: "Jaegar",
    edad: 20
}


// console.log( persona );
// console.table( {persona} );

const persona2 = { ...persona };

persona2.nombre = "Fabio";

console.log( persona );
console.log( persona2 );
















