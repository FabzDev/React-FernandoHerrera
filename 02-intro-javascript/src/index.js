
// function getUserActivo( nombre ){
//     return {
//         uid: "ABC123",
//         username: nombre
//     }
// }

const getUserActivo = ( nombre ) => ({
        uid: "ABC123",
        username: nombre
})


console.log(getUserActivo('Armin'));
