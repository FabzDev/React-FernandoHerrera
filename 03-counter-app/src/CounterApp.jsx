// const plusOne = () => {
//     valor++
// };

import { useState } from "react";

export const CounterApp = ({value}) => {

    const [counter, setCounter] = useState(value);

    

  return (
    <>
      <h1>Counter App</h1>
      <h2>{counter}</h2>
      <button onClick={() => setCounter(counter + 1)}>click</button>
    </>
  );
};
