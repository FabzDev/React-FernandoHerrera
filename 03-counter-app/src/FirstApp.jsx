
// import { Fragment } from "react";

// export const FirstApp = () => {
//   return (
//     <Fragment>
//       <h1>Fabio Escobar</h1>
//       <h2>First App</h2>
//     </Fragment>
//   );
// };

// const getNombre = (nombre) => nombre


// export const FirstApp = (props) => {
export const FirstApp = ({nombre, title}) => {

  return (
    <>
      <h1>Anime App</h1>
      <h2>Title: {nombre}</h2>
      <h3>Episode: {title}</h3>
    </>
  );
};

// TODO fix
// FirstApp.defaultProps = {
//     nombre: "Nombre del Anime",
//     title: "Titulo del capitulo"
// }



