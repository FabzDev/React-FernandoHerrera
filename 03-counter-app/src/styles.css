html,body{
    background-color: #21232A;
    color: white;
    font-family: Helvetica, Arial, sans-serif;
    font-size: 1.3rem;
    padding: 30px;
}

button{
    padding: 5px 10px;
    width: 100px;
    margin-right: 5px;
    border-radius: 5px;
    background-color: #e3ebde;
}