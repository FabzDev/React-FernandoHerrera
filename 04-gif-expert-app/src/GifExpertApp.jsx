import { useState } from "react";

// bkUpWx7ebyORXlTH747bYSgrAmAw64BC
export const GifExpertApp = () => {
  const [categories, setCategories] = useState(['Attack On Titan', 'Ichigo 100%'])
  const addAnime = (anime) => () => setCategories([...categories, anime]);


  return (
    <>
      <h1>Gif Expert App</h1>
      {/* <button onClick={() => setCategories([...categories, "Naruto"])}>Add Category</button> */}
      <button onClick={addAnime('Naruto')}>Add Category</button>
      
      <ul>
        {categories.map(category => <li key={category}>{category}</li>)}
      </ul>    
    </>
  );
};
