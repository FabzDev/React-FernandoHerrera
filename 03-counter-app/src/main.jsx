import React from "react";
import ReactDOM from "react-dom/client"
import { HelloWorldApp } from "./HelloWorldApp";
import { FirstApp } from "./FirstApp";
import './styles.css';
import { CounterApp } from "./CounterApp";


ReactDOM.createRoot( document.getElementById("root"))
.render(
    <React.StrictMode>
        {/* <FirstApp nombre="Inuyasha!" title="Seshomaru ataca de nuevo..."/> */}
        <CounterApp value={10} />
    </React.StrictMode>    
)