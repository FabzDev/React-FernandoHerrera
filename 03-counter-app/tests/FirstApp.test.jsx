import { render } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";

describe("Tests on FirstApp Component", () => {
  // test('should match with the snapshot', () => {
  //   const titleTest = 'Inuyasha!';
  //   const titleTest2 = 'Seshomaru ataca de nuevo...';
  //   const {container} = render( <FirstApp nombre={ titleTest } title={ titleTest2 }/>)
  //   expect(container).toMatchSnapshot();
  // })

  test('should contain "Title: Inuyasha!"', () => {
    const titleTest = "Inuyasha!";
    const titleTest2 = "Seshomaru ataca de nuevo...";
    const { container, getByText } = render(
      <FirstApp nombre={titleTest} title={titleTest2} />
    );
    expect(getByText("Title: Inuyasha!")).toBeTruthy();
  });

//   test("should contain Title: Inuyasha! in a h1 tag", () => {
//     const titleTest = "Title: Inuyasha!";
//     const { container, getByText } = render(
//       <FirstApp nombre={"Inuyasha!"} title={"Seshomaru ataca de nuevo..."} />
//     );
//     const h1 = document.querySelector("h2");
//     expect(h1.innerHTML).toContain(titleTest);
//   });

  test("should contain Title: Inuyasha! in a h1 tag", () => {
    const titleTest = "Anime App";
    const { container, getByTestId } = render(
      <FirstApp nombre={"Inuyasha!"} title={"Seshomaru ataca de nuevo..."} />
    );
    const h1 = getByTestId("h1-anime");
    expect(h1.innerHTML).toContain(titleTest);
  });

  test("should contain title sent by props", () => {
    const titleProp = "Seshomaru ataca de nuevo...";
    const { container, getByTestId } = render(
      <FirstApp nombre={"Inuyasha!"} title={titleProp} />
    );
    
    expect(getByTestId("h3-title").innerHTML).toContain(titleProp);
  });

});
