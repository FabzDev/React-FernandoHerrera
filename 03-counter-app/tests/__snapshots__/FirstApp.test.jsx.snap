// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on FirstApp Component should match with the snapshot 1`] = `
<div>
  <h1>
    Anime App
  </h1>
  <h2>
    Title: 
    Inuyasha!
  </h2>
  <h3>
    Episode: 
    Seshomaru ataca de nuevo...
  </h3>
</div>
`;
